src/rules/fileHeaderRule.ts

```diff
@@ -32,7 +33,9 @@ export class Rule extends Lint.Rules.AbstractRule {
             The second argument, which is optional, is a string that should be inserted as a header comment
             if fixing is enabled and no header that matches the first argument is found.
             The third argument, which is optional, is a string that denotes whether or not a newline should
-            exist on the header.`,
+            exist on the header.
+            The fourth argument, which is optional, is a string that denotes whether or not a header is
+            allowed to consist of consecutive single-line comments.`,
         options: {
```

Comment:
I know this is a bit of a pain, but for the same reason as a few other PRs, we should really switch this rule to allowing an options object. Strings directly in the options list can sometimes intersect with package names that happen to be the same.

`importBlacklistRule.ts` is a good example of one that allows either an array of items or a options object.

Reply:
Yeah, this area is a little tricky... The suggestion is to allow two forms of configuration:

1. [string, string, string] // the current one without `enforce-trailing-newline`
2. [object] // a new form that has existing options as well as `enforce-trailing-newline`

Starting text suggestion for the documentation:

```text
A single object may be passed in for configuration that must contain:

* `match`: a regular expression that all headers should match

Any of the following optional fields may also be provided:

* `allow-single-line-comments`: whether `//` should be considered file headers in additional to `/*` comments
* `default`: text to add for file headers when running in `--fix` mode
* `enforce-trailing-newline`: a boolean for whether a newline must follow the header

The rule will also accept array of strings as a legacy form of options, though the object form is recommended. <insert existing description here>
```

Reply:
Got it.